scripts/run-bundled-codegen.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+# Exit on all errors, undeclared variables and pipefailures.
+set -euo pipefail
+
+# Get the path to the script directory
+SCRIPT_DIR="$(dirname $0)"
+
+# Get the SHASUM of the tarball
+ZIP_FILE="${SCRIPT_DIR}/apollo.tar.gz"
+SHASUM="$(/usr/bin/shasum -a 256 ${ZIP_FILE})"
+SHASUM_FILE="${SCRIPT_DIR}/apollo/.shasum"
+APOLLO_DIR="${SCRIPT_DIR}"/apollo
+
+# Helper functions
+remove_existing_apollo() {
+  rm -r "${APOLLO_DIR}"
+}
+
+extract_cli() {
+  tar xzf "${SCRIPT_DIR}"/apollo.tar.gz -C "${SCRIPT_DIR}"
+  echo "${SHASUM}" | tee "${SHASUM_FILE}"
+}
+
+check_shasum() {
```

Comment:
This function does a bit more than checking the shasum, so could be named more aptly

Reply:
renamed a positively Obj-Cian `validate_codegen_and_extract_if_needed`